From 0635d02aa3323a23aa645a6b0d3e97efda8b7da3 Mon Sep 17 00:00:00 2001
From: Richard Bradley <richard.bradley@joyent.com>
Date: Wed, 11 Oct 2017 08:47:42 +0000
Subject: [PATCH] MANTA-3273 "manta-adm zk fixup" broken with VMs in remote DCs
 Reviewed by: Dave Pacheco <dap@joyent.com> Approved by: Dave Pacheco
 <dap@joyent.com>

---
 lib/adm.js     | 3 ++-
 test/tst.zk.js | 2 --
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/lib/adm.js b/lib/adm.js
index 9945601..3c2a117 100644
--- a/lib/adm.js
+++ b/lib/adm.js
@@ -3959,7 +3959,8 @@ maAdm.prototype.auditZkServers = function ()
 
 		serverUuid = zkinstance.params['server_uuid'];
 		if (!self.ma_vms.hasOwnProperty(uuid)) {
-			if (self.ma_cns[serverUuid] !== null) {
+			if (self.ma_cns.hasOwnProperty(serverUuid) &&
+			    self.ma_cns[serverUuid] !== null) {
 				validationErrors.push(new VError(
 				    'nameservice instance "%s": VM appears ' +
 				    'to have been provisioned in this ' +
diff --git a/test/tst.zk.js b/test/tst.zk.js
index bbe3d85..ccc8b84 100644
--- a/test/tst.zk.js
+++ b/test/tst.zk.js
@@ -112,8 +112,6 @@ function runTestCase(testcase, callback)
 				    'server_uuid': server_uuid
 				};
 			}
-		} else {
-			deployed.cns[server_uuid] = null;
 		}
 
 		deployed.instances[svcid].push({
-- 
2.21.0

